File: labs OOP/2.1/C++/Line.cpp
#include "Line.h"
#include <cmath>

Line::Line() : x1(0), y1(0), x2(0), y2(0) {}

Line::Line(double x1, double y1, double x2, double y2)
	: x1(x1), y1(y1), x2(x2), y2(y2) {}

double Line::calculateLength() const {
	return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));

}
double Line::getX1() const { return x1; }
double Line::getY1() const { return y1; }
double Line::getX2() const { return x2; }
double Line::getY2() const { return y2; }

File: labs OOP/2.1/C#/Line.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Line
{
    protected double x1, y1, x2, y2;

    public Line() { x1 = y1 = x2 = y2 = 0; }

    public Line(double x1, double y1, double x2, double y2)
    {
        this.x1 = x1;
        this.y1 = y1;
        this.x2 = x2;
        this.y2 = y2;
    }

    public double CalculateLength()
    {
        return Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(x2 - x1, 2));

    }
    public double X1 => x1;
    public double Y1 => y1;
    public double X2 => x2;
    public double Y2 => y2;
}
